@eparis @smarterclayton @lavalamp - I am a little confused on what is going on here.  Shouldn't I just have to run `hack/update-all` to regenerate artifacts?  It looks like we are generating stuff that is not liked by `hack/verify-codegen`

see errors like this in travis:

```
2016/04/19 10:33:25 Verifying file "/home/decarr/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/apis/batch/v1/conversion_generated.go"
type checker error: github.com/opencontainers/runc/libcontainer/cgroups/fs/cpuacct.go:22:25: GetClockTicks not declared by package system
type checking encountered some errors in "github.com/opencontainers/runc/libcontainer/cgroups/fs", but ignoring.
```

--- a/api/swagger-spec/v1.json
+++ b/api/swagger-spec/v1.json
@@ -15143,6 +15143,9 @@
    "v1.ConfigMapList": {
     "id": "v1.ConfigMapList",
     "description": "ConfigMapList is a resource containing a list of ConfigMap objects.",
+    "required": [
+     "items"
+    ],
     "properties": {
      "kind": {
       "type": "string",

--- a/docs/api-reference/v1/definitions.html
+++ b/docs/api-reference/v1/definitions.html
@@ -1286,7 +1286,7 @@ Examples:<br>
 <tr>
 <td class="tableblock halign-left valign-top"><p class="tableblock">items</p></td>
 <td class="tableblock halign-left valign-top"><p class="tableblock">Items is the list of ConfigMaps.</p></td>
-<td class="tableblock halign-left valign-top"><p class="tableblock">false</p></td>
+<td class="tableblock halign-left valign-top"><p class="tableblock">true</p></td>
 <td class="tableblock halign-left valign-top"><p class="tableblock"><a href="#_v1_configmap">v1.ConfigMap</a> array</p></td>
 <td class="tableblock halign-left valign-top"></td>
 </tr>
@@ -7742,7 +7742,7 @@ The resulting set of endpoints can be viewed as:<br>
 </div>
 <div id="footer">
 <div id="footer-text">
-Last updated 2016-04-15 09:26:34 UTC
+Last updated 2016-04-21 02:45:03 UTC
 </div>
 </div>
 </body>

--- a/pkg/api/types.generated.go
+++ b/pkg/api/types.generated.go
@@ -46755,14 +46755,13 @@ func (x *ConfigMapList) CodecEncodeSelf(e *codec1978.Encoder) {
 			_, _, _ = yysep2, yyq2, yy2arr2
 			const yyr2 bool = false
 			yyq2[0] = true
-			yyq2[1] = len(x.Items) != 0
 			yyq2[2] = x.Kind != ""
 			yyq2[3] = x.APIVersion != ""
 			var yynn2 int
 			if yyr2 || yy2arr2 {
 				r.EncodeArrayStart(4)
 			} else {
-				yynn2 = 0
+				yynn2 = 1
 				for _, b := range yyq2 {
 					if b {
 						yynn2++
@@ -46802,34 +46801,28 @@ func (x *ConfigMapList) CodecEncodeSelf(e *codec1978.Encoder) {
 			}
 			if yyr2 || yy2arr2 {
 				z.EncSendContainerState(codecSelfer_containerArrayElem1234)
-				if yyq2[1] {
-					if x.Items == nil {
-						r.EncodeNil()
-					} else {
-						yym9 := z.EncBinary()
-						_ = yym9
-						if false {
-						} else {
-							h.encSliceConfigMap(([]ConfigMap)(x.Items), e)
-						}
-					}
-				} else {
+				if x.Items == nil {
 					r.EncodeNil()
+				} else {
+					yym9 := z.EncBinary()
+					_ = yym9
+					if false {
+					} else {
+						h.encSliceConfigMap(([]ConfigMap)(x.Items), e)
+					}
 				}
 			} else {
-				if yyq2[1] {
-					z.EncSendContainerState(codecSelfer_containerMapKey1234)
-					r.EncodeString(codecSelferC_UTF81234, string("items"))
-					z.EncSendContainerState(codecSelfer_containerMapValue1234)
-					if x.Items == nil {
-						r.EncodeNil()
+				z.EncSendContainerState(codecSelfer_containerMapKey1234)
+				r.EncodeString(codecSelferC_UTF81234, string("items"))
+				z.EncSendContainerState(codecSelfer_containerMapValue1234)
+				if x.Items == nil {
+					r.EncodeNil()
+				} else {
+					yym10 := z.EncBinary()
+					_ = yym10
+					if false {
 					} else {
-						yym10 := z.EncBinary()
-						_ = yym10
-						if false {
-						} else {
-							h.encSliceConfigMap(([]ConfigMap)(x.Items), e)
-						}
+						h.encSliceConfigMap(([]ConfigMap)(x.Items), e)
 					}
 				}
 			}

--- a/pkg/api/types.go
+++ b/pkg/api/types.go
@@ -2407,7 +2407,7 @@ type ConfigMapList struct {
 	unversioned.ListMeta `json:"metadata,omitempty"`
 
 	// Items is the list of ConfigMaps.
-	Items []ConfigMap `json:"items,omitempty"`
+	Items []ConfigMap `json:"items"`
 }
 
 // These constants are for remote command execution and port forwarding and are

--- a/pkg/api/v1/types.generated.go
+++ b/pkg/api/v1/types.generated.go
@@ -46567,14 +46567,13 @@ func (x *ConfigMapList) CodecEncodeSelf(e *codec1978.Encoder) {
 			_, _, _ = yysep2, yyq2, yy2arr2
 			const yyr2 bool = false
 			yyq2[0] = true
-			yyq2[1] = len(x.Items) != 0
 			yyq2[2] = x.Kind != ""
 			yyq2[3] = x.APIVersion != ""
 			var yynn2 int
 			if yyr2 || yy2arr2 {
 				r.EncodeArrayStart(4)
 			} else {
-				yynn2 = 0
+				yynn2 = 1
 				for _, b := range yyq2 {
 					if b {
 						yynn2++
@@ -46614,34 +46613,28 @@ func (x *ConfigMapList) CodecEncodeSelf(e *codec1978.Encoder) {
 			}
 			if yyr2 || yy2arr2 {
 				z.EncSendContainerState(codecSelfer_containerArrayElem1234)
-				if yyq2[1] {
-					if x.Items == nil {
-						r.EncodeNil()
-					} else {
-						yym9 := z.EncBinary()
-						_ = yym9
-						if false {
-						} else {
-							h.encSliceConfigMap(([]ConfigMap)(x.Items), e)
-						}
-					}
-				} else {
+				if x.Items == nil {
 					r.EncodeNil()
+				} else {
+					yym9 := z.EncBinary()
+					_ = yym9
+					if false {
+					} else {
+						h.encSliceConfigMap(([]ConfigMap)(x.Items), e)
+					}
 				}
 			} else {
-				if yyq2[1] {
-					z.EncSendContainerState(codecSelfer_containerMapKey1234)
-					r.EncodeString(codecSelferC_UTF81234, string("items"))
-					z.EncSendContainerState(codecSelfer_containerMapValue1234)
-					if x.Items == nil {
-						r.EncodeNil()
+				z.EncSendContainerState(codecSelfer_containerMapKey1234)
+				r.EncodeString(codecSelferC_UTF81234, string("items"))
+				z.EncSendContainerState(codecSelfer_containerMapValue1234)
+				if x.Items == nil {
+					r.EncodeNil()
+				} else {
+					yym10 := z.EncBinary()
+					_ = yym10
+					if false {
 					} else {
-						yym10 := z.EncBinary()
-						_ = yym10
-						if false {
-						} else {
-							h.encSliceConfigMap(([]ConfigMap)(x.Items), e)
-						}
+						h.encSliceConfigMap(([]ConfigMap)(x.Items), e)
 					}
 				}
 			}

--- a/pkg/api/v1/types.go
+++ b/pkg/api/v1/types.go
@@ -2856,7 +2856,7 @@ type ConfigMapList struct {
 	unversioned.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
 	// Items is the list of ConfigMaps.
-	Items []ConfigMap `json:"items,omitempty" protobuf:"bytes,2,rep,name=items"`
+	Items []ConfigMap `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
 
 // Type and constants for component health validation.

